services/catalog/src/simcore_service_catalog/models/domain/service.py

```diff
@@ -0,0 +1,300 @@
+    )
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Structure(BaseModel):
+    key: Union[str, bool, float]
+    label: str
+
+    class Config:
+        extra = Extra.forbid
+
+
+class SelectBox(BaseModel):
+    structure: List[Structure] = Field(..., min_items=1)
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Widget(BaseModel):
+    widget_type: WidgetType = Field(
+        ..., alias="type", description="type of the property"
+    )
+    details: Union[TextArea, SelectBox]
+
+    class Config:
+        extra = Extra.forbid
+
+
+class ServiceProperty(BaseModel):
+    display_order: float = Field(
+        ...,
+        alias="displayOrder",
+        description="use this to numerically sort the properties for display",
+        example=[1, -0.2],
+    )
+    label: str = Field(..., description="short name for the property", example="Age")
+    description: str = Field(
+        ...,
+        description="description of the property",
+        example="Age in seconds since 1970",
+    )
+    property_type: constr(regex=PROPERTY_TYPE_RE) = Field(
+        ...,
+        alias="type",
+        description="data type expected on this input glob matching for data type is allowed",
+        example=[
+            "number",
+            "boolean",
+            "data:*/*",
+            "data:text/*",
+            "data:[image/jpeg,image/png]",
+            "data:application/json",
+            "data:application/json;schema=https://my-schema/not/really/schema.json",
+            "data:application/vnd.ms-excel",
+            "data:text/plain",
+            "data:application/hdf5",
+            "data:application/edu.ucdavis@ceclancy.xyz",
+        ],
+    )
+    file_to_key_map: Optional[Dict[FileName, PropertyName]] = Field(
+        None,
+        alias="fileToKeyMap",
+        description="Place the data associated with the named keys in files",
+        example=[{"dir/input1.txt": "key_1", "dir33/input2.txt": "key2"}],
+    )
+    default_value: Optional[Union[str, float, bool, int]] = Field(
+        None, alias="defaultValue", example=["Dog", True]
+    )
+
+    class Config:
+        extra = Extra.forbid
+
+
+class ServiceInput(ServiceProperty):
+    widget: Optional[Widget] = Field(
+        None,
+        description="custom widget to use instead of the default one determined from the data-type",
+    )
+
+
+class ServiceOutput(ServiceProperty):
+    pass
+
+
+class ServiceKeyVersion(BaseModel):
+    key: constr(regex=KEY_RE) = Field(
+        ...,
+        title="",
+        description="distinctive name for the node based on the docker registry path",
+        # regex=KEY_RE,
+        example=[
+            "simcore/services/comp/itis/sleeper",
+            "simcore/services/dynamic/3dviewer",
+        ],
+    )
+    version: constr(regex=VERSION_RE) = Field(
+        ...,
+        description="service version number",
+        # regex=VERSION_RE,
```

Comment:
maybe you forgot to enable the regex here?

Reply:
no, it's done by the constr. I will remove that comment. good catch.